src/saveload/saveload.h

```diff
@@ -692,11 +690,8 @@ struct SaveLoad {
  */
 #define SLE_CONDNULL(length, from, to) {SL_ARR, SLE_FILE_U8 | SLE_VAR_NULL, length, from, to, 0, nullptr, 0, nullptr}
 
-/** Translate values ingame to different values in the savegame and vv. */
-#define SLE_WRITEBYTE(base, variable) SLE_GENERAL(SL_WRITEBYTE, base, variable, 0, 0, SL_MIN_VERSION, SL_MAX_VERSION, 0)
-
-#define SLE_VEH_INCLUDE() {SL_VEH_INCLUDE, 0, 0, SL_MIN_VERSION, SL_MAX_VERSION, 0, [] (void *b, size_t) { return b; }, 0}
-#define SLE_ST_INCLUDE() {SL_ST_INCLUDE, 0, 0, SL_MIN_VERSION, SL_MAX_VERSION, 0, [] (void *b, size_t) { return b; }, 0}
+/** Only write byte during saving; never read it during loading. */
```

Comment:
The comment is misleading as the byte is read, just separately using `SlReadByte` to ensure the right (sub)class is constructed. Now it reads like: "we write a byte to the save game, that we never need", and I start wondering. Why write a byte to the save game that is never read?

Reply:
I am happy to read that you read it like that, as that is exactly what it does. And sure, we have an intended usage of it, but that is just that: how we intend someone to use it .. but nothing is enforcing that.

It really is a shitty mechanism that is very easy to misunderstand and fuck up. So I am more looking into how to remove it completely ;)

That said, I can add a line of doxygen comment to explain the intended use of it .. but at least you now understood correctly what it did :) The initial comment was truly misleading ;)